### PR TITLE
feat: Cost of carry integration

### PR DESCRIPTION
**Pull request by 0xonramp**
Leverages new API for cost of carry, closed prior PR (#1308)

### CODE REVIEW

File: src/app/leverage/provider.tsx [hunk 1]
```diff
@@ -235,12 +238,21 @@ export function LeverageProvider(props: { children: any }) {
...
+
+  useEffect(() => {
+    if (inputToken === null || outputToken === null) return
 
-    const jsonRpcProvider = publicClientToProvider(publicClient)
     const inputOutputToken = isMinting ? outputToken : inputToken
-    fetchCostOfCarry(jsonRpcProvider, inputOutputToken, setCostOfCarry)
-  }, [publicClient, isMinting, inputToken, outputToken])
+    if (carryCosts)
+      setCostOfCarry(carryCosts[inputOutputToken.symbol.toLowerCase()] ?? null)
```
Comment (0xonramp) [inline src/app/leverage/provider.tsx:254]: @janndriessen one item we have now as part of this UI that we didn't before is ETH2x on mainnet - does the cost of carry still apply there?

File: package.json [hunk 2]
```diff
@@ -18,9 +18,6 @@
   },
   "dependencies": {
     "@0xarc-io/analytics": "2.2.3",
-    "@aave/contract-helpers": "1.28.1",
-    "@aave/math-utils": "1.28.1",
-    "@bgd-labs/aave-address-book": "2.27.0",
```
Comment (janndriessen) [inline package.json:23]: 🧹

File: src/app/leverage/components/leverage-widget/components/fees.tsx [hunk 3]
```diff
@@ -83,10 +83,10 @@ export function Fees(props: FeesProps) {
             />
             <FeesItem label='Mint Fee' percent={'0.10%'} valueUsd={''} />
             <FeesItem label='Redeem Fee' percent={'0.10%'} valueUsd={''} />
-            {/* {props.costOfCarry !== null && (
+            {props.costOfCarry !== null && (
               <FeesItem
                 label='Cost of Carry'
-                showPositiveFee={props.costOfCarry > 0}
+                showPositiveFee={props.costOfCarry < 0}
```
Comment (janndriessen) [inline src/app/leverage/components/leverage-widget/components/fees.tsx:89]: This change is on purpose? Just making sure.

File: src/app/leverage/provider.tsx (hunk 1 above)
Comment (janndriessen) [reply src/app/leverage/provider.tsx:254]: Haven't heard of it there before, so I guess can be omitted for now. Best to check with caf when she's back.

**Review by janndriessen — APPROVED** [review]
So clean 🌟

**Pull request merged**